### ISSUE TITLE
cross domain issues

### ISSUE THREAD

**Issue by QualityWorks** [opencart/opencart#3025]
I have opened this before but it was closed without a resolution.

if you install the shop on a domain www.domain.com and access the shop thorugh domain.com (without www) there are several issues on the initial page you load. 
1) the add to cart button will not work (or any javascript that's being ran from common.js for that matter)
2) font-awesome icons will show as blank squares

this has to do with the loading of the resources cross domain (www to non-www or reverse, since the issue is both ways)

now, the problem is solved as soon as any link is clicked, the links work redirect to the www. domain (or the non-www, depending on where the shop is installed) and on the subsequent page, the buttons and icons work. Due to caching, it will even work on the previous page if you hit back.

So the solution is to do a check somewhere in the core for the browser address and the installed address and redirect if applicable. 

for now, I am using htaccess to fix this for my customers but keep in mind that in the default install, the htaccess is not used, so this issue will remain, and also, the htaccess needs to be written depening on the installation on www or non-www. An actual core check would be the only real option here.

In the last discussion there had been some voices saying these issues don't even exist, all I can say is, check it out. I have now had them on ALL the 2.0 installations I have done, spread over several hosts. (roughly 30 to date)

**Comment by QualityWorks** [comment]
I do see a crossdomain.xml in the root, but it doesn't seem to fix this issue

**Comment by danielkerr** [comment]
you were told the solution. this is not an opencart issue! its how browsers work issue.

**Comment by QualityWorks** [comment]
It’s not even close to a browser issue… it MAY be an apache issue, but it’s
not even that. Just because 2 guys didn’t understand the issue doesn’t make
it go away. Test it yourself.



*Van:* Daniel Kerr [mailto:notifications@github.com]
*Verzonden:* maandag 18 mei 2015 18:26
*Aan:* opencart/opencart
*CC:* Taco
*Onderwerp:* Re: [opencart] cross domain issues (#3025)



you were told the solution. this is not an opencart issue! its how browsers
work issue.

—
Reply to this email directly or view it on GitHub
<https://github.com/opencart/opencart/issues/3025#issuecomment-103118682>.

**Comment by georgipetrov** [comment]
Hi Taco,

This is a default browser behavior called same-domain or same-origin
policy. This is part of every modern browser.  It is one of the security
pivots in the client-server architecture of the internet we know. You can
learn more here http://en.wikipedia.org/wiki/Same-origin_policy

Your very same website should not be accessible on two domains on first
place (browsers regard subdomains as different domains). You should choose
one of the domains, and do an SEO-friendly redirect to the other domain,
and this should be the one of your website. I can assume all your websites
are by default reachable on www.example.com and example.com but this is
because of the standard config your host is doing, by pointing an A record
to both domains. This to big extend minimizes their support efforts of
people constantly asking why their website is not reachable with or without
www. Having two domains with the same site also may have some negative
impact on Google, unless you set this up in their webmasters tools. More
information here https://support.google.com/webmasters/answer/66359?hl=en

In short - choose your preferred domain and stick to it. Make the other to
redirect to the preferred one. Config OpenCart to use the preferred one.
Then it will all work for you, the proper way it should. This is all.

-- Georgi

**Comment by QualityWorks** [comment]
hi Georgi,
I agree with everything you say, and I understand it.
now here comes my customer relations point of view.
Like you said, most hosts set up the hosting with both accesspoints.
so if the package loads through both, and seems to be working through both, but actually doesn't, unless you access through the accesspoint you initially setup, that's a bug. Either it works or it doesn't, inform the users that you need to stick to the one accesspoint, or make it work for both (like 1.5 did)
right now it's confusing..
